File: mobile/lib/features/profile/data/models/organization_model.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'organization_model.freezed.dart';
part 'organization_model.g.dart';

/// Mirrors backend `OrganizationDto`.
/// `verifiedAtUtc == null` means the pending-verification badge.
@freezed
class OrganizationModel with _$OrganizationModel {
  const factory OrganizationModel({
    required String id,
    required String ownerUserId,
    required String name,
    required String type,
    required List<String> causeTags,
    required String location,
    required String description,
    DateTime? verifiedAtUtc,
    required bool isVerified,
  }) = _OrganizationModel;

  factory OrganizationModel.fromJson(Map<String, dynamic> json) =>
      _$OrganizationModelFromJson(json);
}

extension OrganizationModelX on OrganizationModel {
  bool get isPendingVerification => verifiedAtUtc == null && !isVerified;
}

File: frontend/src/app/features/profiles/models/organization.model.ts
/**
 * Mirrors `OrganizationDto` returned by `/api/organizations/*`
 * (camelCase via ASP.NET JSON). `verifiedAtUtc === null` means the
 * organization is pending verification (Phase 1 shows a badge only).
 */
export interface Organization {
  readonly id: string;
  readonly ownerUserId: string;
  readonly name: string;
  readonly type: string;
  readonly causeTags: readonly string[];
  readonly location: string;
  readonly description: string;
  /** ISO-8601 UTC timestamp, or null when pending verification. */
  readonly verifiedAtUtc: string | null;
  readonly isVerified: boolean;
}

export interface OrganizationPayload {
  readonly name: string;
  readonly type: string;
  readonly causeTags: readonly string[];
  readonly location: string;
  readonly description: string;
}
